### PR TITLE
Add "_aligned_malloc" on windows platform

### LINKED ISSUE #953: Add "_aligned_malloc" on windows platform
Hi,

I'm trying to add windows support for a another library which uses `posix_memalign`.  I have researched to the point of realizing that I need libc to expose `_aligned_malloc` for windows.

I'm willing to have a go at a PR to add it but I'm a little out of my depth...

Could someone more knowledgeable point my in the direction of a similar function already implemented in libc and I will try to have a go adding this one.

Obviously, any other advise is welcome.

Thanks

### PR DESCRIPTION
**Pull request by kellda**
Fixes #953

### CODE REVIEW

File: src/windows/mod.rs [hunk 1]
```diff
@@ -529,6 +529,8 @@ extern "C" {
         category: ::c_int,
         locale: *const wchar_t,
     ) -> *mut wchar_t;
+    #[link_name = "_aligned_malloc"]
+    pub fn aligned_malloc(size_t size, size_t alignment) -> *mut c_void;
```
Comment (JohnTitor) [inline src/windows/mod.rs:533]: The syntax is wrong.

Comment (kellda) [reply src/windows/mod.rs]: Oops I copied this straight from the docs... I fixed it now

**Comment by JohnTitor** [comment]
r=me once the commits are squashed.

**Comment by kellda** [comment]
Done. Wondering how much
> If any changes to this PR are deemed necessary, please add them as extra commits.
is applicable...

**Pull request merged**